### PR TITLE
vendor containers/common@main

### PR DESCRIPTION
**Pull request by vrothberg**
The `IgnorePlatform` options has been removed from the
`LookupImageOptions` in libimage to properly support multi-arch images.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
Merged in c/common and c/buildah. Ready from my side.

@containers/podman-maintainers PTAL

**Comment by mheon** [comment]
Restarted one test, LGTM

**Comment by vrothberg** [comment]
> $GOSRC/$SCRIPT_BASE/setup_environment.sh
/tmp/scripts81855ad8681d0d86d1e91e00167939cb.sh: line 5: /var/tmp/go/src/github.com/containers/podman/./contrib/cirrus/setup_environment.sh: No such file or directory

2nd time in this PR. Really looks like a CI infra fart.

**Comment by vrothberg** [comment]
Argh... the buds tests need massaging. I'll be out on Monday.

@edsantiago, do you know what's going on?  In case not, please don't bother and I'll take a look on Tuesday morning.

**Comment by edsantiago** [comment]
This is almost certainly fallout from https://github.com/containers/buildah/pull/3292

TL;DR something changed in runc (I think?) and changed the debug log messages. The fix in the above PR seems to have fixed things for new-runc, but perhaps our podman CI systems are still using old-runc, and the fix breaks.

Solution: make a fix that handles both old-runc and new-runc. Or: bump up podman CI systems to use new-runc.

@cevich PTAL

**Comment by cevich** [comment]
> Solution: make a fix that handles both old-runc and new-runc. Or: bump up podman CI systems to use new-runc.

and wait, there's more!  I'm having trouble bringing in new images due to [some deep/dark SELinux issue only Dan and Giuseppe are qualified to describe](https://github.com/containers/podman/issues/10817) :cry:

**Comment by cevich** [comment]
Update: [Since the update here merged](https://github.com/containers/buildah/pull/3374), maybe try rebasing and see if it helps?

**Comment by vrothberg** [comment]
Still failing:
```
[+0766s] not ok 211 bud with --runtime and --runtime-flag
[+0766s] # (from function `expect_output' in file tests/helpers.bash, line 317,
[+0766s] #  in test file tests/bud.bats, line 2899)
[+0766s] #   `expect_output --substring "$flag_accepted_rx"' failed
```

**Comment by cevich** [comment]
I have those [same new images for this repo](#10829), but it's failing horribly in CI :cry:

**Comment by vrothberg** [comment]
@containers/podman-maintainers @rhatdan please merge me.  This will unblock merging c/common and buildah but note that the bud tests is now skipped until the CI images have been updated.
